src/lib.rs

```diff
@@ -76,4 +76,27 @@ impl RustBoy {
     pub fn get_mem_at(&self, loc: u16) -> u8 {
         self.cpu.get_mem_at(loc)
     }
-}
\ No newline at end of file
+}
+
+use std::os::raw::c_char;
+use std::ffi::{c_void, CStr};
+
+#[no_mangle]
+pub extern fn rustBoyCreate(cartridge_path: *const c_char, save_file_path: *const c_char) -> *const c_void {
+
+	let save_path_result = unsafe { CStr::from_ptr(save_file_path) };
+	let save_path = match save_path_result.to_str() {
```

Comment:
Something you can do here is `save_path_result.to_str().expect("Failed to parse save path");`. Which is a bit cleaner if you are just panicing if there is an error.

Reply:
Ideally I would like to return null if there is an error, but I'm not sure how.

Reply:
This https://doc.rust-lang.org/std/ptr/fn.null.html seems to be what you are looking for. So you could do

```
match cart_path_result.to_str() {
	Ok(c) => c,
	Err(_) => {
             println!("Failed to parse cartridge path");
             return std::ptr::null;
        }
};
```

Reply:
It would be nice if we could return a result type. So swift could then handle the error case.

Reply:
@coopersimon That null type looks like it it exactly what I'm looking for. When you say return in a match, is that within the match or for the function? I'm a bit confused by your example.

Reply:
Return inside a match returns from the function. So this is basically like a `guard let` kind of situation.

Reply:
@coopersimon  Returning null now. I guess it would be more optimal to return an error code or something so you know why it failed, but I think this is ok for now.